components/org.wso2.carbon.identity.api.server.cors/org.wso2.carbon.identity.api.server.cors.v1/src/main/java/org/wso2/carbon/identity/api/server/cors/v1/core/CORSService.java

```diff
@@ -0,0 +1,229 @@
+ * you may not use this file except in compliance with the License.        
+ * You may obtain a copy of the License at                                 
+ *                                                                         
+ * http://www.apache.org/licenses/LICENSE-2.0                              
+ *                                                                         
+ * Unless required by applicable law or agreed to in writing, software     
+ * distributed under the License is distributed on an "AS IS" BASIS,       
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and     
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.identity.api.server.cors.v1.core;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.api.server.common.ContextLoader;
+import org.wso2.carbon.identity.api.server.common.error.APIError;
+import org.wso2.carbon.identity.api.server.common.error.ErrorResponse;
+import org.wso2.carbon.identity.api.server.cors.common.CORSServiceHolder;
+import org.wso2.carbon.identity.api.server.cors.common.Constants;
+import org.wso2.carbon.identity.api.server.cors.v1.function.CORSOriginToCORSOriginGetObject;
+import org.wso2.carbon.identity.api.server.cors.v1.model.CORSOriginGetObject;
+import org.wso2.carbon.identity.api.server.cors.v1.model.CORSOriginPatch;
+import org.wso2.carbon.identity.api.server.cors.v1.model.CORSOriginPatchObject;
+import org.wso2.carbon.identity.cors.mgt.core.exception.CORSManagementServiceClientException;
+import org.wso2.carbon.identity.cors.mgt.core.exception.CORSManagementServiceException;
+import org.wso2.carbon.identity.cors.mgt.core.exception.CORSManagementServiceServerException;
+import org.wso2.carbon.identity.cors.mgt.core.model.CORSOrigin;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import javax.ws.rs.core.Response;
+
+/**
+ * Call internal OSGi services to perform server CORS management.
+ */
+public class CORSService {
+
+    private static final Log log = LogFactory.getLog(CORSService.class);
+
+    /**
+     * Get a list of associated applications of a CORS origin.
+     *
+     * @return List of associated applications.
+     */
+    public List<String> getAssociatedAppsByCORSOrigin(String corsOriginId) {
+
+        try {
+            String tenantDomain = ContextLoader.getTenantDomainFromContext();
+            List<CORSOrigin> corsOriginList = CORSServiceHolder.getInstance().getCorsManagementService()
+                    .getTenantCORSOrigins(tenantDomain);
+            List<String> associatedApps = new ArrayList<>();
+            for (CORSOrigin corsOrigin : corsOriginList) {
+                if (corsOrigin.getId().equals(corsOriginId)) {
```

Comment:
Any possibility of `corsOrigin.getId()` being null? If so `StringUtils.equals()` can be used to avoid the possible NPE.